app/src/main/java/com/g/tragosapp/ui/cocktaildetails/CocktailsDetailFragment.kt

```diff
@@ -46,11 +52,64 @@ class CocktailsDetailFragment : Fragment() {
         binding.cocktailTitle.text = cocktail.name
         binding.cocktailDesc.text = cocktail.description
 
-        binding.btnSaveCocktail.setOnClickListener {
-            viewModel.saveCocktail(FavoritesEntity(cocktail.cocktailId,cocktail.image,cocktail.name,cocktail.description,cocktail.hasAlcohol))
-            Toast.makeText(requireContext(), "Cocktail saved to favorites", Toast.LENGTH_SHORT)
-                .show()
-        }
+        viewModel.getFavoriteCocktails().observe(viewLifecycleOwner, Observer { result ->
+            when (result) {
+                is Resource.Loading -> {
+                }
+                is Resource.Success -> {
+                    when (result.data.contains(cocktail)) {
+                        true -> {
+                            binding.btnSaveCocktail.setOnClickListener {
+                                viewModel.deleteCocktail(
+                                    FavoritesEntity(
+                                        cocktail.cocktailId,
+                                        cocktail.image,
+                                        cocktail.name,
+                                        cocktail.description,
+                                        cocktail.hasAlcohol
+                                    )
+                                ).observe(viewLifecycleOwner, Observer {
+                                    if (it is Resource.Success) {
+                                        context?.shortToast("Drink deleted !")
+                                        findNavController().navigate(
```

Comment:
Is necessary to reload the fragment after deleting the cocktail? I think the line below that changes the button drawable should be enought since the viewModel handles the deletion and save of the cocktails.

Reply:
ARE YOU STUPID OR GETTING UP ??